xformers/components/attention/csrc/cuda/attention.cu

```diff
@@ -241,25 +330,36 @@ __device__ void compute_loop(
     scalar_t m_prime[kBlockSizeQ],
     scalar_t s_prime[kBlockSizeQ],
     vec_t buffer[kBlockSizeQ][BUFFER_SIZE] /*TODO [BUFFER_SIZE limitation]*/,
-    int64_t K) {
+    int64_t K,
+    scalar_t* attn_bias_i,
+    at::PhiloxCudaState philox_args,
+    int64_t global_offset,
+    int64_t N,
+    scalar_t p,
+    int64_t col_offset) {
   scalar_t si[kBlockSizeQ][kBlockSizeK] = {0};
   compute_dot<scalar_t, vec_t, kBlockSizeK, kBlockSizeQ>(
       query_block, key_i, si, K);
 
+  if (attn_bias_i != nullptr) {
+    add_attn_bias<scalar_t, vec_t, kBlockSizeK, kBlockSizeQ>(si, attn_bias_i);
+  }
+
   scalar_t m_i[kBlockSizeQ];
   compute_max<scalar_t, kBlockSizeK, kBlockSizeQ>(si, m_prime, m_i);
 
   scalar_t m_delta[kBlockSizeQ];
   scalar_t s_delta[kBlockSizeQ][kBlockSizeK];
 
-  compute_scaling_coeffs<scalar_t, kBlockSizeK, kBlockSizeQ>(
-      m_i, m_prime, si, m_delta, s_delta);
+  compute_and_update_scaling_coeffs<scalar_t, kBlockSizeK, kBlockSizeQ>(
+      m_i, m_prime, s_prime, si, m_delta, s_delta);
+
+  if (p < 1.0)
```

Comment:
p > 0. checked elsewhere I suppose ? hard to see with the PR, sorry if dumb question

Reply:
ah wait, in that case the computation is dispatched to the original Flash implementation I guess

Reply:
ahh, it's `p = 1. - p`, aka p keep... all good